src/Provider/Provider.Web/Orchestrators/SearchHelperOrchestrator.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Client;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore.Projections;
+
+namespace Esfa.Recruit.Provider.Web.Orchestrators
+{
+    public class SearchHelperOrchestrator
+    {
+        const int maxCount = 50;
+        private readonly IProviderVacancyClient _providerVacancyClient;
+        public SearchHelperOrchestrator(IProviderVacancyClient providerVacancyClient)
+        {
+            _providerVacancyClient = providerVacancyClient;
+        }
+        public async Task<IEnumerable<string>> GetAutoCompleteListAsync(string term, long ukprn)
+        {
+            var result = new List<string>();
+
+            if (term == null || term.Trim().Length < 3) return result;
+
+            var vacancies = await GetVacanciesAsync(ukprn);
+
+            var data = vacancies
```

Comment:
This fails when I try and test your PR, if I try to search by vacancy reference number for one that is in my list e.g. `VAC1000000042` then the `vacancies` variable has retrieved all the vacancies. The vacancy reference number isn't in the title so the code on line 26 would essentially be an empty Enumerable<string> I would have thought but instead it's type is `System.Linq.Enumerable.EnumerablePartition<string>`

Then when this tries to get evaluated for `data.Count()` in the `if` condition it throws a null reference exception causing the ajax response on the webpage to receive a 500 response.

Reply:
Tried that, unable to reproduce the error. You will have to give me exact steps to reproduce this. I saw the EnumerablePartition but Count didn't throw any error.

Reply:
I get it everytime using the steps I have described above. Can someone else test it?

Reply:
The data.Count is gone now as part of Pete's suggestion.